sound/soc/sof/fw-file-profile.c

```diff
@@ -0,0 +1,311 @@
+	if (base_profile->tplg_path)
+		out_profile->tplg_path = base_profile->tplg_path;
+	else
+		out_profile->tplg_path = desc->default_tplg_path[ipc_type];
+
+	/* topology name */
+	out_profile->tplg_name = plat_data->tplg_filename;
+
+	out_profile->ipc_type = ipc_type;
+
+	/* Test only default firmware file */
+	if (!base_profile->fw_path && !base_profile->fw_name)
+		ret = sof_test_firmware_file(dev, out_profile, NULL);
+
+	if (!ret)
+		ret = sof_test_topology_file(dev, out_profile);
+
+out:
+	if (ret) {
+		/* Free up path strings created with devm_kasprintf */
+		if (fw_path_allocated)
+			devm_kfree(dev, out_profile->fw_path);
+		if (fw_lib_path_allocated)
+			devm_kfree(dev, out_profile->fw_lib_path);
+
+		memset(out_profile, 0, sizeof(*out_profile));
+	}
+
+	return ret;
+}
+
+static void
+sof_missing_firmware_notification(struct snd_sof_dev *sdev,
+				  enum sof_ipc_type ipc_type,
+				  struct sof_loadable_file_profile *base_profile)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	const struct sof_dev_desc *desc = plat_data->desc;
+	struct device *dev = sdev->dev;
+	char *marker;
+	int i;
+
+	dev_err(dev, "SOF firmware and/or topology file not found.\n");
+	dev_info(dev, "Supported default profiles\n");
+
+	for (i = 0; i < SOF_IPC_TYPE_COUNT; i++) {
+		if (!(desc->ipc_supported_mask & BIT(i)))
+			continue;
+
+		if (i == ipc_type)
+			marker = "Requested";
+		else
+			marker = "Fallback";
+
+		dev_info(dev, "- ipc type %d (%s):\n", i, marker);
+		if (base_profile->fw_path_postfix)
+			dev_info(dev, " Firmware file: %s/%s/%s\n",
+				 desc->default_fw_path[i],
+				 base_profile->fw_path_postfix,
+				 desc->default_fw_filename[i]);
+		else
+			dev_info(dev, " Firmware file: %s/%s\n",
+				 desc->default_fw_path[i],
+				 desc->default_fw_filename[i]);
+
+		dev_info(dev, " Topology file: %s/%s\n",
+			 desc->default_tplg_path[i],
+			 plat_data->tplg_filename);
+	}
+
+	if (base_profile->fw_path || base_profile->fw_name ||
+	    base_profile->tplg_path || base_profile->tplg_name)
+		dev_info(dev, "Verify the path/name override module parameters.\n");
+
+	dev_info(dev, "Check if you have 'sof-firmware' package installed.\n");
+	dev_info(dev, "Optionally it can be manually downloaded from:\n");
+	dev_info(dev, "   https://github.com/thesofproject/sof-bin/\n");
+}
+
+static void sof_print_profile_info(struct snd_sof_dev *sdev,
+				   enum sof_ipc_type ipc_type,
+				   struct sof_loadable_file_profile *profile)
+{
+	struct device *dev = sdev->dev;
+
+	if (ipc_type != profile->ipc_type)
+		dev_info(dev,
+			 "Using fallback IPC type %d (requested type was %d)\n",
+			 profile->ipc_type, ipc_type);
+
+	dev_dbg(dev, "Loadable file profile for ipc type %d:\n", profile->ipc_type);
+
+	dev_dbg(dev, " Firmware file:     %s/%s\n", profile->fw_path,
+		profile->fw_name);
+
+	if (profile->fw_lib_path)
+		dev_dbg(dev, " Firmware lib path: %s\n", profile->fw_lib_path);
+
+	dev_dbg(dev, " Topology file:     %s/%s\n", profile->tplg_path,
```

Comment:
Can you please increase the log levels when logging the loaded files? To be honest this is something `request_firmware()` itself should do (imagine that: a "bootloader" that does not even log what it boots) but I guess this is unlikely to happen any time soon.

Some more context in
- #3867

Reply:
I don't think normal users care about this at all and just going to make the kernel log noisy for no benefit.

Adding a debugfs file would be a better solution if we need it (separate files or one file which prints the paths and names?)

Reply:
Define "normal users". Anyone staring at kernel logs and _not_ caring what firmware and configuration files are in use is seriously shooting themselves in the foot.

The kernel squeals "TAINTED" as soon as something is slightly out of the ordinary but somehow loading unknown files from random locations (`/lib/firmware/updates/`, 
`/lib/firmware/UTS_RELEASE/`, etc.) does not deserve a word.

And please don't get me started on the (lack of) security for which logging is one of the most basic requirements.

>  make the kernel log noisy for no benefit.

In "normal" usage, files are loaded only ONCE per boot so this would be obviously not noisy at all.

It would be printed more often only when running a test like "kmod-unload-load" but then such logs become even more relevant.